core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -591,6 +587,18 @@ public static ByteSizeValue parseByteSize(String s, ByteSizeValue minValue, Byte
         return value;
     }
 
+    public static Setting<ByteSizeValue> memorySizeSetting(String key, ByteSizeValue value, Property... properties) {
```

Comment:
can we get some javadocs here and also document what the differences are to the `byteSizeSetting` methods